test/checkr/api_class_test.rb

```diff
@@ -48,6 +48,46 @@ class ApiClassTest < Test::Unit::TestCase
           :filter => 'test filter',
         })
       end
+
+      should 'allow passing an api_key as a param when the global api_key is set' do
+        response = test_response(test_mock_resource_list)
+        api_key = '123456'
+
+        @mock.expects(:get).with do |url, headers, params|
+          (url == "#{Checkr.api_base}#{MockResource.path}?page=1&filter=test%20filter" ||
+            url == "#{Checkr.api_base}#{MockResource.path}?filter=test%20filter&page=1") &&
+            params == nil &&
+            Checkr.api_key == 'foo'
+            headers["Authorization"] == "Basic #{Base64.encode64("#{api_key}:")}"
+        end.returns(response)
+
+        list = MockResource.all(
+          :page => 1,
+          :filter => 'test filter',
+          :api_key => api_key
+        )
+      end
+
+      should 'allow passing an api_key as a param when the global api_key is NOT set' do
+        response = test_response(test_mock_resource_list)
+        api_key = '123456'
+
+        @mock.expects(:get).with do |url, headers, params|
+          (url == "#{Checkr.api_base}#{MockResource.path}?page=1&filter=test%20filter" ||
+            url == "#{Checkr.api_base}#{MockResource.path}?filter=test%20filter&page=1") &&
+            params == nil &&
+            Checkr.api_key == nil &&
```

Comment:
I would put this on a different `expect` since it's not really related to the `get` call.